### DISCUSSION TITLE
Known Camera Poses Reconstruction Pipeline

### DISCUSSION THREAD

**Discussion by mokrueger** [colmap/colmap#1247] [Q&A]
Hi,

I'm currently trying to implement a reconstruction pipeline with known camera poses. 
However I've been struggling a bit with finding the right steps / right ordering of the steps.

I have images and poses from an AR Application.

Would this pipeline work:
1. Feature extraction (As this doesnt use poses, right?)
2. Exhaustive matcher (Also does not use poses(Or does it?)
3. Extract ImageID information from of the database (as we need this to construct the images.txt properly)
4. Create Images.txt, cameras.txt, points3d.txt according to faq
5. Point triangulator with manually created sparse model
6. Maybe bundle_adjuster
6. Image undistorter
7. patch match stereo
8. stereo fusion

Also I noticed theres a "prior" database entry.
Do i also need to fill the poses from the cameras.txt into the database priors?
What step uses these prior values?

Could anyone give me an overview which steps use which information.

Thanks in advance. :)

**Comment by shubham-monarch** [comment]
I am not sure but might want to checkout `model_aligner`.

**Comment by jared-krauss** [comment]
Did you ever find an answer to this?

What were you trying to achieve?

**Reply by mokrueger** [reply]
@jared-krauss I think I ended up trying to use the PointTriangulator with Bundle Adjustment. So pretty much what I described in the question.

**Comment by pablospe** [comment]
Check this answer in the FAQ:
https://colmap.github.io/faq.html#reconstruct-sparse-dense-model-from-known-camera-poses
